server/src/main/java/org/opensearch/extensions/ExtensionRequest.java

```diff
@@ -24,41 +25,60 @@
  */
 public class ExtensionRequest extends TransportRequest {
     private static final Logger logger = LogManager.getLogger(ExtensionRequest.class);
-    private ExtensionsManager.RequestType requestType;
+    private final ExtensionsManager.RequestType requestType;
+    private final String uniqueId;
 
     public ExtensionRequest(ExtensionsManager.RequestType requestType) {
+        this(requestType, null);
+    }
+
+    public ExtensionRequest(ExtensionsManager.RequestType requestType, String uniqueId) {
         this.requestType = requestType;
+        this.uniqueId = uniqueId;
     }
 
     public ExtensionRequest(StreamInput in) throws IOException {
         super(in);
         this.requestType = in.readEnum(ExtensionsManager.RequestType.class);
+        this.uniqueId = in.readOptionalString();
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeEnum(requestType);
+        out.writeOptionalString(uniqueId);
     }
 
     public ExtensionsManager.RequestType getRequestType() {
         return this.requestType;
     }
 
+    @Nullable
+    public String getUniqueId() {
+        return uniqueId;
```

Comment:
Consider avoiding null here by making this:
```java
public Optional<String> getUniqueId() {
    return uniqueId == null ? Optional.empty() : Optional.of(uniqueId);
}
```
Or, consider handling the `readOptionalString(in)` similarly to the above and storing the value internally as an `Optional<String>`, which after looking at the `equals()` implementation is a better idea.

Reply:
Since we set uniqueId as `Optional<String>`, it should be `return uniqueId == null ? Optional.empty() : uniqueId;`